samza-core/src/main/java/org/apache/samza/coordinator/CoordinationService.java

```diff
@@ -0,0 +1,34 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.samza.coordinator;
+
+import org.apache.samza.coordinator.leaderelection.LeaderElector;
+
+
+public interface CoordinationService {
+  // life cycle
+  void start();
```

Comment:
If there isn't any implementation for these method , can we please remove them until we have a solid reason to add it? From what I understand, this pluggable service doesn't enforce the user to use both leaderElector and latch implementations. So, in such a case, adding "start" to initialize certain environment doesn't seem correct. Will it be initialized with respect to leader elector or processor? 
LeaderElector and Latch are their own interfaces, anyway. What is the need to put them together in a "CoordinationService" abstraction? 

Perhaps, @xinyuiscool can help you answer that question